cryptographic_estimators/SDEstimator/sd_helper.py

```diff
@@ -20,92 +20,111 @@
+        precision (int): The number of decimal places to truncate the input value `x`.
 
+    Returns:
+        float: The truncated value of `x`.
     """
+    pass
 
-    return float(int(x * 10 ** precision) / 10 ** precision)
+    return float(int(x * 10**precision) / 10**precision)
 
 
-def __round_or_truncate_to_given_precision(T: float, M: float, truncate: bool, precision: int):
-    """
-    rounds or truncates the inputs `T`, `M`
-    INPUT:
-        - ``T`` -- first value to truncate or round
-        - ``M`` -- second value to truncate or round
-        - ``truncate`` -- if set the `true` the inputs are truncated otherwise rounded
-        - ``precision`` -- precision of the truncation or rounding
+def __round_or_truncate_to_given_precision(T: float, M: float, truncate: bool, precision: int) -> tuple:
+    """Rounds or truncates the input values `T` and `M`.
+
+    Args:
+        T (float): The first value to truncate or round.
+        M (float): The second value to truncate or round.
+        truncate (bool): If set to `True`, the inputs are truncated. Otherwise, they are rounded.
+        precision (int): The number of decimal places to round or truncate the values to.
+
+    Returns:
+        Tuple[str, str]: The rounded or truncated values of `T` and `M` as strings, formatted to the specified precision.
     """
     if truncate:
         T, M = __truncate(T, precision), __truncate(M, precision)
     else:
         T, M = round(T, precision), round(M, precision)
-    return '{:.{p}f}'.format(T, p=precision), '{:.{p}f}'.format(M, p=precision)
+    return "{:.{p}f}".format(T, p=precision), "{:.{p}f}".format(M, p=precision)
 
 
 def _gaussian_elimination_complexity(n: int, k: int, r: int):
-    """
-    Complexity estimate of Gaussian elimination routine
-
-    INPUT:
-
-    - ``n`` -- Row additions are perfomed on ``n`` coordinates
-    - ``k`` -- Matrix consists of ``n-k`` rows
-    - ``r`` -- Blocksize of method of the four russian for inversion, default is zero
+    """Compute the complexity estimate of the Gaussian elimination routine.
 
-    [Bar07]_ Bard, G.V.: Algorithms for solving linear and polynomial systems of equations over finite fields
-    with applications to cryptanalysis. Ph.D. thesis (2007)
+    Args:
+        n (int): The number of rows on which row additions are performed.
+        k (int): The number of rows in the matrix.
+        r (int): The block size of the method of the four Russian for inversion. The default value is zero.
 
-    [BLP08] Bernstein, D.J., Lange, T., Peters, C.: Attacking and defending the mceliece cryptosystem.
-    In: International Workshop on Post-Quantum Cryptography. pp. 31–46. Springer (2008)
+    Returns:
+        The complexity estimate of the Gaussian elimination routine.
 
-    EXAMPLES::
-
-        sage: from cryptographic_estimators.SDEstimator import _gaussian_elimination_complexity
-        sage: _gaussian_elimination_complexity(n=100,k=20,r=1) # random
+    References:
```

Comment:
citation issue

Reply:
Which is the error, sorry? The changed format (year position, pages instead of pp., etc)?

Reply:
@FloydZ could you please add the comment to the actual problematic line in the future, like this i always have to go to the file to check it 😇

Reply:
what floyd means is that citations have been removed. The citations [Bar07]_ [BLP08]_ should be mentioned. But the docstring was already inconsistent before

Reply:
I am always marking the problematic position, but admittedly, the message in this case is not particularly good. I think the problem is, that if someone applies changes to this file after I commented, the line numbers are misaligned, which leads to this misaligned references.
Is is possible to mark multiple lines in such a comment? That would make it a lot easier.

Reply:
@FloydZ Yes, you can. Click the first line of the block, then press shift, and click the end line of the block.